### DISCUSSION TITLE
Why is there a sendLpp method in lpsTdoa2Tag.c

### DISCUSSION THREAD

**Discussion by danteppc** [bitcraze/discussions#404] [Q&A] [answered]
Hello,

I am not sure I understand why [sendLppShort](https://github.com/bitcraze/crazyflie-firmware/blob/57357f101934a1abc49aa31cb170f25a3addeb2a/src/deck/drivers/src/lpsTdoa2Tag.c#L157) method belongs to a TDoA source file. I also see almost identical functions in [lpsTdoa3Tag.c](https://github.com/bitcraze/crazyflie-firmware/blob/57357f101934a1abc49aa31cb170f25a3addeb2a/src/deck/drivers/src/lpsTdoa3Tag.c#L207) and [lpsTwrTag.c](https://github.com/bitcraze/crazyflie-firmware/blob/ecf9a0b8548df3ac5722a58ed20baf8d7675ddbe/src/deck/drivers/src/lpsTwrTag.c#L338). Especially since there's no need for a TX on the tag side in TDoA mode. I can only assume this is really just to update the positions from PC since Crazyradio cannot directly update the anchors? if so, would it be possible to find a better place for this than in a TDoA implementation? 
Thanks

**Comment by ataffanel** [comment, marked as answer]
Hi,

These function are used to communicate with the anchor in order to configure them, the main use-case is to configure the anchor position when setting up a new system, but other things like the ranging mode can be configured as well. You are correct that in TDoA mode there is no communication needed from the Crazyflie to the anchor, so this is a special case used only for configuration.

The files where these functions are implemented are responsible for the TDoA and TWR air-protocol implementation, so I think these function are actually well placed as they implement part of the air protocol for LPP for the different protocols. The actual TDoA implementation (the math of it) and LPP message handling are implemented  centrally in other files.

ps. I converted your issue into a discussion topic since this is a better fit for questions.
